tests/rules/method.js

```diff
@@ -219,6 +220,31 @@ eslintTester.run("method", rule, {
             code: "(e = node.insertAdjacentHTML('beforebegin', '<s>safe</s>'))()",
             parserOptions: { ecmaVersion: 6 },
         },
+        // Typescript support tests
+        {
+            code: "return !!selection && (!this.options.isVisible || (this.options.isVisible as any)(selection as any, ...args))",
+            parser: PATH_TO_TYPESCRIPT_ESLINT,
+            parserOptions: {
+                ecmaVersion: 2018,
+                sourceType: 'module',
+            }
+        },
+        {
+            code: "if ((<ParsedStringPattern>parsedPatterns[i])(path, basename)) { return 1 }",
```

Comment:
I think you can trim down the test case. Do we need the if statement here? Do we need a function that accepts two params? Etc.

Reply:
done!